src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystem.java

```diff
@@ -133,37 +134,72 @@ public static void createReactivePowerDistributionEquations(EquationSystem equat
         return qKeys;
     }
 
+    private static void createNonImpedantBranch(VariableSet variableSet, EquationSystem equationSystem,
+                                                LfBranch branch, LfBus bus1, LfBus bus2) {
+        boolean hasV1 = equationSystem.hasEquation(bus1.getNum(), EquationType.BUS_V);
+        boolean hasV2 = equationSystem.hasEquation(bus2.getNum(), EquationType.BUS_V);
+        if (!(hasV1 && hasV2)) {
+            // create voltage magnitude coupling equation
+            // 0 = v1 - v2 * rho
+            PiModel piModel = branch.getPiModel();
+            double rho = piModel.getR2() / piModel.getR1();
+            equationSystem.createEquation(branch.getNum(), EquationType.ZERO_V)
+                    .addTerm(new BusVoltageEquationTerm(bus1, variableSet))
+                    .addTerm(EquationTerm.multiply(new BusVoltageEquationTerm(bus2, variableSet), -1 * rho));
+
+            // add a dummy reactive power variable to both sides of the non impedant branch and with an opposite sign
+            // to ensure we have the same number of equation and variables
+            equationSystem.createEquation(bus1.getNum(), EquationType.BUS_Q)
+                    .addTerm(new DummyReactivePowerEquationTerm(branch, variableSet));
+            equationSystem.createEquation(bus2.getNum(), EquationType.BUS_Q)
+                    .addTerm(EquationTerm.multiply(new DummyReactivePowerEquationTerm(branch, variableSet), -1));
+        } else {
+            // nothing to do in case of v1 and v2 are not found, we just have to ensure
```

Comment:
Are found ?

Reply:
yes, I fixed the comment. Also I realized that I forgot to add a check somewhere to avoid PV buses at both sides of non impedant branch with inconsistent target voltage.